webapp/channels/src/components/autocomplete_selector.tsx

```diff
@@ -1,55 +1,68 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import SuggestionBox from 'components/suggestion/suggestion_box';
 import SuggestionList from 'components/suggestion/suggestion_list';
 
-export default class AutocompleteSelector extends React.PureComponent {
-    static propTypes = {
-        providers: PropTypes.array.isRequired,
-        value: PropTypes.string.isRequired,
-        onSelected: PropTypes.func,
-        label: PropTypes.node,
-        labelClassName: PropTypes.string,
-        inputClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string,
-        footer: PropTypes.node,
-        disabled: PropTypes.bool,
-        toggleFocus: PropTypes.func,
-        listComponent: PropTypes.elementType,
-        listPosition: PropTypes.string,
-    };
+import type ModalSuggestionList from './suggestion/modal_suggestion_list';
+import type Provider from './suggestion/provider';
+
+type Props = {
+    id?: string;
+    providers: Provider[];
+    value: string;
+    onSelected?: (selected: Selected) => void;
+    label?: React.ReactNode | string;
+    labelClassName?: string;
+    inputClassName?: string;
+    helpText?: React.ReactNode | string;
+    placeholder?: string;
+    footer?: Node;
+    disabled?: boolean;
+    toggleFocus?: ((focus: boolean) => void) | null;
+    listComponent?: typeof SuggestionList | typeof ModalSuggestionList;
+    listPosition?: string;
+};
+
+type State = {
+    input: string;
+    focused?: boolean;
+};
+
+type Selected = {
+    id: string;
+    username: string;
+    display_name: string;
+    value: string;
+    text: string;
+}
 
-    static defaultProps = {
```

Comment:
I have mixed feelings about removing the defualtProps. You made the behavior the same in the render, but I wonder if we ever come back to this component to add more functionality, if we will miss the defaults in any other method. 0/5

Reply:
`defaultProps` are added back. I also removed the defaults in render.